sklearn/neighbors/_classification.py

```diff
@@ -707,75 +710,98 @@ def predict_proba(self, X):
             The class probabilities of the input samples. Classes are ordered
             by lexicographic order.
         """
-
+        check_is_fitted(self, "_fit_method")
         n_queries = _num_samples(X)
 
-        neigh_dist, neigh_ind = self.radius_neighbors(X)
-        outlier_mask = np.zeros(n_queries, dtype=bool)
-        outlier_mask[:] = [len(nind) == 0 for nind in neigh_ind]
-        outliers = np.flatnonzero(outlier_mask)
-        inliers = np.flatnonzero(~outlier_mask)
-
-        classes_ = self.classes_
-        _y = self._y
-        if not self.outputs_2d_:
-            _y = self._y.reshape((-1, 1))
-            classes_ = [self.classes_]
+        metric, metric_kwargs = _adjusted_metric(
+            metric=self.metric, metric_kwargs=self.metric_params, p=self.p
+        )
 
-        if self.outlier_label_ is None and outliers.size > 0:
-            raise ValueError(
-                "No neighbors found for test samples %r, "
-                "you can try using larger radius, "
-                "giving a label for outliers, "
-                "or considering removing them from your dataset." % outliers
+        if (
+            self.weights == "uniform"
+            and self._fit_method == "brute"
+            and not self.outputs_2d_
+            and RadiusNeighborsClassMode.is_usable_for(X, self._fit_X, metric)
+        ):
+            probabilities = RadiusNeighborsClassMode.compute(
+                X=X,
+                Y=self._fit_X,
+                radius=self.radius,
+                weights=self.weights,
+                labels=self._y,
+                unique_labels=self.classes_,
+                outlier_label=self.outlier_label,
+                metric=metric,
+                metric_kwargs=metric_kwargs,
+                strategy="parallel_on_X",
```

Comment:
Are there benchmarks to support a fixed dispatch on `X` rather than an arbitrary heuristic?

Reply:
I think this is based on the current call to `ArgKminClassMode`:

https://github.com/scikit-learn/scikit-learn/blob/59048f9821db191eba14bee46e903e602be770a2/sklearn/neighbors/_classification.py#L327-L345

Benchmarks you previously made are reported in https://github.com/scikit-learn/scikit-learn/pull/24076#issuecomment-1445258342.

I think it is worth replicating the comment here as well.

Reply:
Yes I did not run any specific benchmarks with regards to thsi but it does seem that parallel_on_X runs generally better.